infra/cifuzz/config_utils.py

```diff
@@ -88,7 +59,62 @@ def _get_language():
 # pylint: disable=too-few-public-methods,too-many-instance-attributes
 
 
-class ConfigurationError(Exception):
+class CiEnvGetter:
+  """Class for getting environment variables that are CI-specific."""
+
+  # Mapping from generic variable names to GitHub-specific ones. This is used on
+  # GitHub to get the variable values from the environment.
+  GITHUB_ENV_VAR_MAPPING = {
+      'WORKSPACE': 'GITHUB_WORKSPACE',
+      'REPOSITORY': 'GITHUB_REPOSITORY',
+      'TOKEN': 'GITHUB_TOKEN',
+      'GIT_SHA': 'GITHUB_SHA'
+  }
+
+  def __init__(self, is_github):
+    self.is_github = is_github
+
+  def _map_variable(self, variable):
+    # Returns the ci-specific variable name if needed.
+    if self.is_github:
+      return self.GITHUB_ENV_VAR_MAPPING[variable]
+    return variable
+
+  def get(self, variable):
+    """Returns the value of variable in the environment."""
+    return os.getenv(self._map_variable(variable))
+
+  def get_project_repo_owner_and_name(self):
+    """Returns a tuple containing the project repo owner and the name of the
+    repo."""
+    variable = self._map_variable('REPOSITORY')
+    # On GitHub this includes owner and repo name.
+    repository = os.getenv(variable, '')
+    # Use os.path.split. When GITHUB_REPOSITORY just contains the name of the
+    # repo, this will return a tuple containing an empty string and the repo
+    # name. When GITHUB_REPOSITORY contains the repo owner followed by a slash
+    # and then the repo name, it will return a tuple containing the owner and
+    # repo name.
+    return os.path.split(repository)
+
+  def get_project_src_path(self, workspace):
```

Comment:
no need to require a `workspace` variable here, as it's leaking an implementation detail for github env handling. We can just read the workspace again in GitHub's implementation (see above comment).

Reply:
Done.